### PR TITLE
Codegen framework output test

### PR DESCRIPTION
**Pull request by BobaFetters**
- Added a Codegen test configuration to test taking codegen output and generating it into an XCFramework
- Added a new docs page that details general usage of Apollo code as an XCFramework as well as starts compiling tips for things to be aware of collected from apollographql/apollo-ios#3271

Closes apollographql/apollo-ios#3271

### CODE REVIEW

File: docs/source/development/using-xcframework.mdx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+title: Using Apollo as an XCFramework
+---
+
+If you need to use Apoll iOS and the code generation output as an XCFramework, this page will give a high level overview of how to do so, as well as collect some tips on things you may encounter that have been reported by the community.
```
Comment (AnthonyMDev) [inline docs/source/development/using-xcframework.mdx:5]:
```suggestion
If you need to use Apollo iOS and the code generation output as an XCFramework, this page will give a high level overview of how to do so, as well as collect some tips on things you may encounter that have been reported by the community.
```

File: docs/source/config.json [hunk 2]
```diff
@@ -103,7 +103,8 @@
     ],
     "Development & Testing": [
       {
-        "Test Mocks": "/testing/test-mocks"
+        "Test Mocks": "/testing/test-mocks",
+        "Apollo as an XCFramework": "/development/using-xcframework"
```
Comment (AnthonyMDev) [inline docs/source/config.json:107]:
```suggestion
        "Including Apollo as an XCFramework": "/development/using-xcframework"
```

File: docs/source/development/using-xcframework.mdx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Creating the XCFramework
+
+### Apollo iOS
+
+In order to use Apollo iOS as an XCFramework, we provide a [apollo-ios-xcframework]() repository that can generate XCFrameworks from the Apollo iOS code including the necessary commands. This can be used as is or as a starting point for you to customize further for your needs.
```
Comment (AnthonyMDev) [inline docs/source/development/using-xcframework.mdx:11]: Missing the link url to the framework project here.

File: docs/source/development/using-xcframework.mdx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+### Code Generation Output
+
+You can also use the output from code generation as an XCFramework, you will need to configure this setup yourself, however you can use of [code generation test](https://github.com/apollographql/apollo-ios-dev/tree/main/Tests/TestCodeGenConfigurations/CodegenXCFramework) of this as an example/starting point.
```
Comment (AnthonyMDev) [inline docs/source/development/using-xcframework.mdx:15]:
```suggestion
You can also use the output from code generation as an XCFramework, you will need to configure this setup yourself. You can use the [code generation test](https://github.com/apollographql/apollo-ios-dev/tree/main/Tests/TestCodeGenConfigurations/CodegenXCFramework) project as an example/starting point for this.
```

**Review by AnthonyMDev — CHANGES_REQUESTED** [review]
We should probably add all the generated files with the `.graphql.swift` extension [to be marked as generated](https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github). This will automatically collapse them on PRs, and exclude them from contribution statistics.

**Comment by calvincestari** [comment]
> We should probably add all the generated files with the .graphql.swift extension [to be marked as generated](https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github). This will automatically collapse them on PRs, and exclude them from contribution statistics.

I had no idea about that! It's a very nice quality-of-life feature.

**Comment by BobaFetters** [comment]
@AnthonyMDev Added the `.gitattributes` file, not sure if maybe those changes won't take affect for the files in this diff or because they are new? They don't appear to be auto-collapsed in the diff for me like it says they should be.

**Review by calvincestari — APPROVED** [review]
This looks good from my pov. If we can get that collapsed diff trick figured out it'd be great otherwise I'm good with this.

**Comment by calvincestari** [comment]
I see that we've got some warnings showing up in the test codegen config project builds from merging in the defer work. They don't seem to be failing the CI test so I'm not going to act on them just yet. Merging in the rest of defer will clear those up.

**Pull request merged**